### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO LINK](https://AndriiYelieva.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -22,38 +23,49 @@ export const TextField: React.FC<Props> = ({
...
-            'is-danger': hasError,
+            'is-danger': hasError || hasUrlError,
           })}
           placeholder={placeholder}
           value={value}
-          onChange={event => onChange(event.target.value)}
-          onBlur={() => setTouched(true)}
+          name={name}
+          onChange={onChange}
+          onBlur={handleFieldBlur} // Fixed: Removed unnecessary arrow function wrapper
```
Comment (OleziO) [inline src/components/TextField/TextField.tsx:63]: Remove all comments from the code.

File: src/services/checkIsValidUrl.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const checkIsValidUrl = (url: string) => {
+  // eslint-disable-next-line
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
```
Comment (OleziO) [inline src/services/checkIsValidUrl.ts:4]: Split it by a few lines and remove the rule disabling.
image

File: tsconfig.json [hunk 3]
```diff
@@ -3,4 +3,4 @@
   "include": [
     "src"
   ]
-}
+}
```
Comment (OleziO) [inline tsconfig.json:6]: Add an empty line at the end of the file.

**Review by OleziO — CHANGES_REQUESTED** [review]
Add string trim to avoid creating the empty movie.
image

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Check this case
https://github.com/mate-academy/react_movies-list-add-form/assets/61904995/acd9fd2c-b158-44af-923a-95d1726b0530
2023-07-17_15h01_29

**Review by etojeDenys — APPROVED** [review]

**Pull request closed without merging**